src/main/java/frc/robot/RobotContainer.java

```diff
@@ -71,15 +106,60 @@ public RobotContainer() {
+    // Controller buttons
     m_switchCameraButton = new JoystickButton(m_controller1, Constants.SWAPCAMBUTTON);
-    m_aimButton = new JoystickButton(m_flightStick, Constants.AIMBUTTON);
     m_balanceButton = new JoystickButton(m_controller1, Constants.BALANCEBUTTON);
     m_straightButton = new JoystickButton(m_controller1, Constants.STRAIGHTBUTTON);
+    // Joystick buttons
+    m_clawButton = new JoystickButton(m_flightStick, Constants.CLAWBUTTON);
+    m_aimButton = new JoystickButton(m_flightStick, Constants.AIMBUTTON);
+    m_armDownButton = new JoystickButton(m_flightStick, Constants.ARMDOWNBUTTON);
+    m_armUpButton = new JoystickButton(m_flightStick, Constants.ARMUPBUTTON);
     // commands
     m_balanceButton.whileTrue(m_balanceCommand);
     m_aimButton.whileTrue(m_aimCommand);
     m_straightButton.whileTrue(m_straightCommand);
+    m_clawButton.toggleOnTrue(m_clawCommand).toggleOnFalse(m_clawCommand);
+    m_armDownButton.whileTrue(m_armDownCommand);
+    m_armUpButton.whileTrue(m_armUpCommand);
+  }
+
+  private void initializeAutonomous() {
+    // Network Table Routine Options
+    autoDashboardChooser.setDefaultOption("Auto With Balancing", "FullAuto");
+    autoDashboardChooser.addOption("Auto Without Balancing", "AutoNoBalance");
+    autoDashboardChooser.addOption("Auto Test Mode", "Test");
+    SmartDashboard.putData(autoDashboardChooser);
+
+    // Events
+    // ex:
+    // autonomousEventMap.put("A", new PathFollowingCommand(m_driveSubsystem, pathGroup.get(0)));
+    autonomousEventMap.put("BalanceRobot", m_balanceCommand);
+
+    // Create the AutoBuilder. This only needs to be created once when robot code starts, not every
+    // time you want to create an auto command.
+    // Use blue side of field when designing!
+    autoBuilder =
+        new RamseteAutoBuilder(
+            m_driveSubsystem::getPose, // Pose2d supplier
+            m_driveSubsystem
+                ::resetPose, // Pose2d consumer, used to reset odometry at the beginning of auto
+            new RamseteController(
+                DriveConstants.kRamseteB, DriveConstants.kRamseteZeta), // RamseteController
+            DriveConstants.kDriveKinematics, // DifferentialDriveKinematics
+            DriveConstants
+                .FeedForward, // A feedforward value to apply to the drive subsystem's controllers
+            m_driveSubsystem
+                ::getWheelSpeeds, // A method for getting the current wheel speeds of the drive
+            new PIDConstants(
+                DriveConstants.kPDriveVel, 0, 0), // A PID controller for wheel velocity control
+            m_driveSubsystem
+                ::tankDriveVolts, // A consumer that takes left and right wheel voltages and sets
+            // them to the drive subsystem's controllers
+            autonomousEventMap,
+            true, // change for either team
+            m_driveSubsystem //  Requirements of the commands (should be the drive subsystem)
+            );
   }
```

Comment:
Are we resetting the odometry before initializing? ref https://github.com/mjansen4857/pathplanner/wiki/PathPlannerLib:-Java-Usage#ppramsetecommand